src/Microsoft.Restier.Publishers.OData/RestierController.cs

```diff
@@ -76,16 +80,59 @@ private ApiBase Api
                 throw new InvalidOperationException(Resources.ControllerRequiresPath);
             }
 
+            var result = default(IQueryable);
+
+            // Get queryable path builder to builder
             IQueryable queryable = this.GetQuery();
-            QueryRequest queryRequest = new QueryRequest(queryable)
+
+            // If it is a unbound function, call Api function directly
+            ODataPathSegment lastSegment = path.Segments.LastOrDefault();
+            if (lastSegment != null && lastSegment.SegmentKind != ODataSegmentKinds.UnboundFunction)
```

Comment:
Refactor the code in this method (from line 90 to line 133):
 - Too many condition `lastSegment != null`. Actually if it is null, no result will be created (though I even doubt if this case would happen). Thus either throw exception (if necessary) or just remove this condition.
 - Three if(s) (line 90, line 118 and line 128) can be merged to make the code much clearer. The main logic here has three parts: unbound, bound and the others. Suggest we write a switch statement (on `lastSegment.SegmentKind`) with three cases; extract line 92 to line 114 to a private function. Each case just needs to call two to three methods.